enterprise/cmd/frontend/internal/auth/sourcegraphoperator/middleware.go

```diff
@@ -110,10 +111,23 @@ func authHandler(db database.DB) func(w http.ResponseWriter, r *http.Request) {
 			}
 
 			var expiry time.Duration
-			// If the "sourcegraph-operator" is the only external account associated with the
-			// user, that means the user is a pure Sourcegraph Operator which should have
-			// designated and aggressive session expiry.
+			// If the "sourcegraph-operator" (SOAP) is the only external account associated
+			// with the user, that means the user is a pure Sourcegraph Operator which should
+			// have designated and aggressive session expiry - unless that account is designated
+			// a service account.
```

Comment:
```suggestion
			// as a service account.
```

not a native speaker, but doesn't feel right